src/rqt_image_view/image_view.cpp

```diff
@@ -317,10 +317,10 @@ void ImageView::onTopicChanged(int index)
 
   if (!topic.isEmpty())
   {
-    image_transport::ImageTransport it(node_);
     const image_transport::TransportHints hints(node_.get(), transport.toStdString());
     try {
-      subscriber_ = it.subscribe(topic.toStdString(), 1, &ImageView::callbackImage, this, &hints);
+      subscriber_ = image_transport::create_subscription(node_.get(), topic.toStdString(), std::bind(&ImageView::callbackImage, this, std::placeholders::_1), 
+        hints.getTransport(), rmw_qos_profile_sensor_data);
```

Comment:
I think this change is an improvement. I think we can make it even better by allowing QoS configuration via ROS parameters. Similar to https://github.com/ros-perception/image_pipeline/pull/651, I think all we need to do is the following:
```suggestion
      auto subscription_options = rclcpp::SubscriptionOptions();
      subscription_options.qos_overriding_options = rclcpp::QosOverridingOptions::with_default_policies();
      subscriber_ = image_transport::create_subscription(
        node_.get(),
        topic.toStdString(),
        std::bind(&ImageView::callbackImage, this, std::placeholders::_1),
        hints.getTransport(),
        rmw_qos_profile_sensor_data,
        subscription_options);
```
You can find more about QoS overrides in the design doc: https://design.ros2.org/articles/qos_configurability.html

(I've also proposed some line-breaks to avoid one really long line).